### PR TITLE
feat(dev): CTL-2042 — fleet host env-file materialization + doctor drift check

### PR DESCRIPTION
**Pull request by ryanrozich**
## Summary

Five days of fleet deadlock — twice, through two different variables — trace to the same root: `~/.config/catalyst/execution-core.env` is never materialized from the cluster repo, so a hand-edit or a reinstall silently diverges one host without any signal. This PR closes that gap end-to-end.

Three changes land together because they form one coherent gate-and-monitor:

1. **`syncHostEnvFiles()` in `cluster-sync.mjs`** — materializes `hosts/<host>/execution-core.env` from the cluster repo into `~/.config/catalyst/` at 0o600 on boot (`clusterSync`) and on every periodic refresh (`refreshClusterSecretsIfChanged`). Its `changed[]` output is merged into the existing restart-required emitter, so a rotated `execution-core.env` triggers the same operator signal as a rotated secret file.

2. **`execution-core-secrets.env` in SECRET_REGISTRY** — registered as `env-file/boot-only` in both `secret-contract.mjs` and `catalyst-secret-contract.sh`. This is the SOPS-encrypted overlay (carrying `CATALYST_WORKFLOW_GITHUB_TOKEN`) that is sourced at the end of the committed plain posture file. Splitting the non-secret posture (commit-plain, reviewable in `git diff`) from the credential (stays SOPS-encrypted) closes the half of the gap that sync alone can't cover.

3. **`checkExecutionCoreEnvDrift` in `doctor.mjs`** — a new advisory (`STATUS.WARN`, never `FAIL`) doctor check that compares `~/.config/catalyst/execution-core.env` against the committed cluster repo copy. Reports variable **names** that differ, never values. Wired into `execution-core-env-drift-health.mjs` (208 lines, new) which is the testable leaf the check delegates to.

### `plugins/dev/scripts/execution-core/cluster-sync.mjs` (+129/-5)

- `syncHostEnvFiles(clusterRepoRoot, hostName, destDir?)`: resolves `<clusterRepoRoot>/hosts/<hostName>/execution-core.env`, reads it, writes to `~/.config/catalyst/execution-core.env` at mode 0o600. Returns `{changed: string[], errors: string[]}`.
- Called from `clusterSync()` (boot path) and `refreshClusterSecretsIfChanged()` (periodic path); `changed[]` merged into the restart-required set alongside existing secret-file changed arrays.
- `syncHostEnvFiles` is fail-soft: a missing host-specific env file is not an error (single-host or new hosts have none yet).

### `plugins/dev/scripts/execution-core/execution-core-env-drift-health.mjs` (NEW, +208)

- `checkEnvDrift(onDiskPath, repoPath)`: reads both files, extracts variable names (ignoring comments and blank lines), returns `{drifted: string[], onDiskOnly: string[], repoOnly: string[]}`.
- `renderDriftReport(result)`: formats the drift summary for `catalyst doctor` output.
- Pure functions — no side effects, no file writes. Injected paths, no hardcoded `~/.config/catalyst`.

### `plugins/dev/scripts/execution-core/doctor.mjs` (+6/-0)

- Wires `checkExecutionCoreEnvDrift` as a new advisory check in the doctor's check sequence.
- Grade: `STATUS.WARN` when drift detected, `STATUS.OK` when in sync, `STATUS.SKIP` when either file is absent (not yet deployed, single-host).

### `plugins/dev/scripts/lib/secret-contract.mjs` (+12/-0) and `catalyst-secret-contract.sh` (+12/-12)

- Adds `execution-core-secrets.env` row to `SECRET_REGISTRY`: type `env-file`, rotation class `boot-only`, presence-checked but value never read by the contract engine.
- Parity suite (`secret-contract-parity.test.sh`) updated: 179/179 pass.

### Test coverage

| File | Added | Notes |
|------|-------|-------|
| `cluster-sync.test.mjs` | +196 | 83 cases covering `syncHostEnvFiles` — missing file, unchanged, changed, mode enforcement, path resolution |
| `execution-core-env-drift-health.test.mjs` | +354 | 33 cases covering `checkEnvDrift` edge cases: empty files, comments, blank lines, partial overlaps, identical |
| `secret-contract.test.mjs` | +12/-3 | Covers new `execution-core-secrets.env` row |
| `secret-contract-parity.test.sh` | +3/-3 | Updated row counts for parity suite |

### CI

- `.github/workflows/execution-core-tests.yml`: adds `execution-core-env-drift-health.test.mjs` to the stable test list.

## Test plan

- [x] `bun test execution-core/cluster-sync.test.mjs` — 83 pass, 0 fail
- [x] `bun test lib/secret-contract.test.mjs` — 115 pass, 0 fail
- [x] `bun test execution-core/execution-core-env-drift-health.test.mjs` — 33 pass, 0 fail
- [x] `bash __tests__/secret-contract-parity.test.sh` — 179/179 pass
- [x] `bun test execution-core/doctor.test.mjs` — 443 pass, 13 fail (identical pre-existing baseline failures, not in this diff)

## Related Issues/PRs

- Fixes https://linear.app/coalesce-labs/issue/CTL-2042

### CODE REVIEW

File: plugins/dev/scripts/execution-core/cluster-sync.mjs [hunk 1]
```diff
@@ -854,9 +968,13 @@ export function refreshClusterSecretsIfChanged(opts = {}) {
   // CTL-1595: the direnv-profile bundle rides the same refresh; profilesDir
   // intentionally NOT overridable from here (tests inject via clusterSync/opts).
   const profiles = syncProfileFiles({ clusterDir, profilesDir: opts.profilesDir, ageKeyFile, decrypt: sopsDecrypt, logger });
+  // CTL-2042: per-host plain posture file materialization (non-SOPS, readable in git diff).
+  const _syncHostEnvFilesImpl = typeof opts.syncHostEnvFiles === "function" ? opts.syncHostEnvFiles : syncHostEnvFiles;
+  const hostEnvFiles = _syncHostEnvFilesImpl({ clusterDir, configDir, logger });
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/cluster-sync.mjs:1074]: Fixed in 6f3369fc. Added `gitHostEnvChangedBetween` and, in `refreshClusterSecretsIfChanged`, when `secrets/` is unchanged but `hosts/<host>/` changed we now materialize the posture (pinned to the same host identity as the diff) and emit restart-required before advancing the marker; a write shortfall keeps the marker behind so the next tick retries. Covered by cluster-sync.test.mjs (b2)/(b3).

File: plugins/dev/scripts/execution-core/cluster-sync.mjs [hunk 2]
```diff
@@ -484,6 +484,114 @@ export function syncProfileFiles(opts = {}) {
...
+// on write errors (records in failed[], does not throw). An absent hosts/ directory
+// or no file for this host → skipped: true.
+//
+// Result shape mirrors syncSecretFiles():
+//   { written[], changed[], failed[], skipped, reason }
+export function syncHostEnvFiles(opts = {}) {
+  const {
+    clusterDir = getClusterRepoDir(),
+    configDir = defaultConfigDir(),
+    hostName: hostNameOpt,
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/cluster-sync.mjs:513]: Fixed in 6f3369fc. `syncHostEnvFiles` now resolves the write target from `CATALYST_EXECUTION_CORE_ENV` (the launcher's canonical env-file path) when set, else the configDir default, creating the parent dir as needed; the tracked dest name stays the logical basename so the restart-required filter is unaffected. The doctor drift check resolves its disk path the same way. Covered by the new 'writes to the CATALYST_EXECUTION_CORE_ENV override path' test.

File: plugins/dev/scripts/execution-core/execution-core-env-drift-health.mjs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  if (diskError) {
+    return { verdict: VERDICT.INCONCLUSIVE, reason: `disk: ${diskError.code}` };
+  }
+  if (repoError) {
+    return { verdict: VERDICT.INCONCLUSIVE, reason: `repo: ${repoError.code}` };
+  }
+
+  // Repo absent → this host has no committed posture; nothing to drift from.
+  if (repoContent === null) {
+    return { verdict: VERDICT.ABSENT, onlyOnDisk: [], onlyInRepo: [], valueDiffers: [] };
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/execution-core-env-drift-health.mjs]: Fixed in 6f3369fc. Added a positive control: `checkExecutionCoreEnvDrift` verifies the cluster clone + `hosts/` tree exist and threads `clusterAvailable` into the classifier; a null repo read from an un-inspectable source now returns INCONCLUSIVE (WARN) rather than a false-clean ABSENT. Covered by the new 'cluster clone is unavailable (positive control)' test.

File: plugins/dev/scripts/execution-core/execution-core-env-drift-health.mjs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  if (r.onlyOnDisk?.length) parts.push(`only on disk: ${r.onlyOnDisk.join(", ")}`);
+  if (r.onlyInRepo?.length) parts.push(`only in repo: ${r.onlyInRepo.join(", ")}`);
+  if (r.valueDiffers?.length) parts.push(`value differs: ${r.valueDiffers.join(", ")}`);
+  const nameSummary = parts.length ? parts.join("; ") : "content differs";
+
+  return mkCheck(
+    "execution-core-env-drift",
+    STATUS.WARN,
+    `execution-core.env DRIFT detected for host ${String(hostName)}. Variable names: ${nameSummary}. ` +
+      `Run \`catalyst-stack cluster-sync\` or wait for the refresh timer to materialize the committed posture. ` +
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/execution-core-env-drift-health.mjs]: Fixed in 6f3369fc. The drift remediation message now points at `catalyst cluster sync` (the implemented verb in execution-core/cli/cluster.mjs), not the nonexistent `catalyst-stack cluster-sync`.

**Comment by ryanrozich** [comment]
@codex review
